src/main/java/com/google/devtools/build/lib/skyframe/SingleToolchainResolutionFunction.java

```diff
@@ -139,7 +140,8 @@ private static SingleToolchainResolutionValue resolveConstraints(
       }
 
       // Make sure the target platform matches.
-      if (!checkConstraints(
+      if (!toolchain.targetSettings().stream().allMatch(ConfigMatchingProvider::matches)
```

Comment:
We should probably enhance the output of --toolchain_resolution_debug to clarify when a toolchain is rejected for settings instead of target/exec constraints.

Reply:
Done.